app/src/pages/playground/PlaygroundDatasetExamplesTable.tsx

```diff
@@ -82,7 +349,128 @@ export function PlaygroundDatasetExamplesTable({
 }: {
   datasetId: string;
 }) {
+  const environment = useRelayEnvironment();
   const instances = usePlaygroundContext((state) => state.instances);
+  const setExperimentId = usePlaygroundContext(
+    (state) => state.setExperimentId
+  );
+  const [dialog, setDialog] = useState<ReactNode>(null);
+
+  const hasSomeRunIds = instances.some(
+    (instance) => instance.activeRunId !== null
+  );
+
+  const credentials = useCredentialsContext((state) => state);
+  const markPlaygroundInstanceComplete = usePlaygroundContext(
+    (state) => state.markPlaygroundInstanceComplete
+  );
+  const playgroundStore = usePlaygroundStore();
+
+  const [exampleResponsesMap, setExampleResponsesMap] =
+    useState<InstanceToExampleResponsesMap>(
+      getInitialExampleResponsesMap(instances)
+    );
+  const notifyError = useNotifyError();
+
+  const onNext = useCallback(
+    (instanceId: number) =>
+      (response?: PlaygroundDatasetExamplesTableSubscription$data | null) => {
+        if (response == null) {
+          return;
+        }
+        const chatCompletion = response.chatCompletionOverDataset;
+        switch (chatCompletion.__typename) {
+          case "ChatCompletionSubscriptionError":
+            markPlaygroundInstanceComplete(instanceId);
+            notifyError({
+              title: "Chat completion failed",
+              message: chatCompletion.message,
+              expireMs: 10000,
+            });
+            break;
```

Comment:
setExperimentId(null)?

Reply:
i need to fix this error handling but i don't think we want to mark instances as complete nor remove experimentId here. We should map each error message to it's appropriate cell.